src/python/pants/init/target_roots_calculator.py

```diff
@@ -127,23 +127,38 @@ def create(cls, options, session, symbol_table, build_root=None):
     # initialization paths.
     changed_options = options.for_scope('changed')
     changed_request = ChangedRequest.from_options(changed_options)
-
+    owned_files = options.for_global_scope().owner_of
     logger.debug('spec_roots are: %s', spec_roots)
     logger.debug('changed_request is: %s', changed_request)
     scm = get_scm()
     change_calculator = ChangeCalculator(session, symbol_table, scm) if scm else None
+    owner_calculator = OwnerCalculator(session, symbol_table) if owned_files else None
+    logger.debug('owner_files are: %s', owned_files)
+    targets_specified = [1 for spec
+                         in (changed_request.is_actionable(), owned_files, spec_roots)
+                         if spec]
+
+    if len(targets_specified) > 1:
+      # We've been provided a more than one of: a change request, an owner request, or spec roots.
+      raise InvalidSpecConstraint(
+        'Only one target specification can be provided out of these three options: '
+        '--changed-*, --owner-of=, or a specific target'
+      )
```

Comment:
It'd be nice if this said what specifications were provided that conflicted, but I don't think it's blocking. It might be a bit tricky to re-ify the cli representations.

Also, I think the wording here is a little jargony--target specification isn't a term that's consistently used in a user facing way. Also, options is a bit overloaded since some of the ways to specify targets are options and some aren't. How about something like this?

`Multiple target selection methods provided. Please use only one of --changed-*, --owner-of, or explicit target arguments`

Also, if there isn't a test that covers this error case, it'd be good to add one.

Reply:
As to the conflicting specifications, I think saying which 3 methods conflict will let the user know why did the command they just entered failed. Although I think it would make for a better error message, I think the added value would be marginal. What do you think?

As to the wording of the error message, I did have a bit of trouble wording it right, thank you for your experienced opinion in what words to use and how. I'll change the error message.

And yes, I'm adding the test! 👽